orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy

```diff
@@ -123,6 +124,19 @@ class OperationsController {
     return planPipeline(pipelineConfig, resolveArtifacts)
   }
 
+  /**
+   * Used by echo to mark an execution failure if it fails to materialize the pipeline
+   * (e.g. because the artifacts couldn't be resolved)
+   *
+   * @param pipeline pipeline json
+   */
+  @RequestMapping(value = '/fail', method = RequestMethod.POST)
+  void failPipeline(@RequestBody Map pipeline) {
+    String errorMessage = pipeline.remove("errorMessage")
```

Comment:
How come you're calling `remove` here?

Reply:
I didn't want to pollute the pipeline object since the error will be persisted as a `cancellationReason` by `markPipelineFailed` anyway